client/src/app/tabs/form/FormEditor.js

```diff
@@ -288,6 +289,43 @@ export class FormEditor extends CachedComponent {
     }
   }
 
+  handleDataEditorInteraction(fn) {
+    const {
+      onAction
+    } = this.props;
+
+    const dataEditorNode = this.ref.current.querySelector('.cfp-data-container');
+
+    if (!dataEditorNode) {
+      return;
+    }
+
+    let data;
+
+    const handleInteractionStart = () => {
+      data = this.getInputData();
+    };
+
+    const handleInteractionEnd = () => {
+      const newData = this.getInputData();
+
+      // fire event once data was touched (changed)
+      if (data !== newData) {
+        onAction('emit-event', {
+          type: 'form.modeler.inputDataChanged'
+        });
+      }
+    };
+
+    if (fn === 'on') {
+      dataEditorNode.addEventListener('focusin', handleInteractionStart);
+      dataEditorNode.addEventListener('focusout', handleInteractionEnd);
+    } else {
+      dataEditorNode.removeEventListener('focusin', handleInteractionStart);
+      dataEditorNode.removeEventListener('focusout', handleInteractionEnd);
+    }
```

Comment:
Since the handlers are declared inline, they will not be unsubscribed correctly.

Reply:
So `removeEventListener` will be called with other callback than `addEventListener`